lib/matplotlib/axes/_axes.py

```diff
@@ -5832,11 +5832,16 @@ def imshow(self, X, cmap=None, norm=None, *, aspect=None,
             which can be set by *filterrad*. Additionally, the antigrain image
             resize filter is controlled by the parameter *filternorm*.
 
-        interpolation_stage : {'data', 'rgba'}, default: 'data'
-            If 'data', interpolation
-            is carried out on the data provided by the user.  If 'rgba', the
-            interpolation is carried out after the colormapping has been
-            applied (visual interpolation).
+        interpolation_stage : {'auto', 'data', 'rgba'}, default: 'auto'
+            If 'data', interpolation is carried out on the data provided by the user,
+            useful if interpolating between pixels during upsampling.
+            If 'rgba', the interpolation is carried out in RGBA-space after the
+            color-mapping has been applied, useful if downsampling and combining
+            pixels visually. The default 'auto' is appropriate for most
+            applications where 'rgba' is used when downsampling, or upsampling at a
+            rate less than 3, and 'data' is used when upsampling at a higher rate.
+            See :doc:`/gallery/images_contours_and_fields/image_antialiasing` for
+            a discussion of image antialiasing.
```

Comment:
Optional: I like bullet-listing
```suggestion
            Supported values:
            
            - 'data': Interpolation is carried out on the data provided by the user
              This is useful if interpolating between pixels during upsampling.
            - 'rgba': The interpolation is carried out in RGBA-space after the
              color-mapping has been applied. This is useful if downsampling and
              combining pixels visually.
            - 'auto': Select a suitable interpolation stage automatically. This uses
              'rgba' when downsampling, or upsampling at a rate less than 3, and
              'data' when upsampling at a higher rate.
              
            See :doc:`/gallery/images_contours_and_fields/image_antialiasing` for
            a discussion of image antialiasing.
```